jiant/tasks/tasks.py

```diff
@@ -2994,6 +2994,95 @@ def get_metrics(self, reset=False):
         return {"accuracy": acc}
 
 
+@register_task("hellaswag", rel_path="HellaSwag/")
+class HellaSwagTask(MultipleChoiceTask):
+    """ Task class for HellaSwag.  """
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        super().__init__(name, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data_text = None
+
+        self.scorer1 = CategoricalAccuracy()
+        self.scorers = [self.scorer1]
+        self.val_metric = "%s_accuracy" % name
+        self.val_metric_decreases = False
+        self.n_choices = 4
+
+    def load_data(self):
+        """ Process the dataset located at path.  """
+
+        def _load_split(data_file):
+            questions, choicess, targs, idxs = [], [], [], []
+            data = [json.loads(l) for l in open(data_file, encoding="utf-8")]
+            for example in data:
+                sent1 = tokenize_and_truncate(
+                    self._tokenizer_name, example["ctx_a"], self.max_seq_len
+                )
+                questions.append(sent1)
+                sent2_prefix = example["ctx_b"]
+                choices = [
+                    tokenize_and_truncate(
+                        self._tokenizer_name, sent2_prefix + " " + ending, self.max_seq_len
+                    )
+                    for ending in example["endings"]
+                ]
+                choicess.append(choices)
+                targ = example["label"] if "label" in example else 0
+                idx = example["ind"]
+                targs.append(targ)
+                idxs.append(idx)
+            return [questions, choicess, targs, idxs]
+
+        self.train_data_text = _load_split(os.path.join(self.path, "hellaswag_train.jsonl"))
+        self.val_data_text = _load_split(os.path.join(self.path, "hellaswag_val.jsonl"))
+        self.test_data_text = _load_split(os.path.join(self.path, "hellaswag_test.jsonl"))
+        self.sentences = (
+            self.train_data_text[0]
+            + self.val_data_text[0]
+            + [choice for choices in self.train_data_text[1] for choice in choices]
+            + [choice for choices in self.val_data_text[1] for choice in choices]
+        )
+        log.info("\tFinished loading HellaSwag data.")
+
+    def process_split(
+        self, split, indexers, model_preprocessing_interface
+    ) -> Iterable[Type[Instance]]:
+        """ Process split text into a list of AlleNNLP Instances. """
```

Comment:
Another typo

Reply:
Yeah, typo proliferates through copy and paste. Sorry, I'll fix it.